File: lib/models/usuario.dart
import './paciente.dart';
import './medico.dart';

class Usuario {
  final int id;
  final String nombre;
  final String correo;
  final String rol;
  final Paciente? paciente;
  final Medico? medico;

  Usuario({
    required this.id,
    required this.nombre,
    required this.correo,
    required this.rol,
    this.paciente,
    this.medico,
  });

  factory Usuario.fromJson(Map<String, dynamic> json) {
    return Usuario(
      id: json['usuario_id'] ?? 0,
      nombre: json['nombre'] ?? '',
      correo: json['correo'] ?? '',
      rol: json['rol'] ?? 'paciente',
      paciente: json.containsKey('paciente') ? Paciente.fromJson(json['paciente']) : null,
      medico: json.containsKey('medico') ? Medico.fromJson(json['medico']) : null,
    );
  }
}

File: android/app/src/main/kotlin/com/example/mi_app/database/Usuario.kt
package com.example.mi_app.database

import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "Usuario",
    indices = [Index(value = ["correo"], unique = true)]
)
data class Usuario(
    @PrimaryKey(autoGenerate = true)
    val usuario_id: Int = 0,
    val nombre: String,
    val correo: String,
    val contrasena: String, // En una app real, esto debería ser un hash
    val rol: String, // "Paciente", "Médico", "Investigador"
    val fecha_creacion: String,
    val activo: Boolean = true
)
